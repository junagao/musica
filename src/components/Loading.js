import React from 'react';

import './Loading.scss';

const Loading = () => (
  <div className="spinner-container">
    <div className="spinner">
      <span className="spinner-inner-1" />
      <span className="spinner-inner-2" />
      <span className="spinner-inner-3" />
    </div>
  </div>
);

export default Loading;
