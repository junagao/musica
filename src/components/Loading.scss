@import '../variables';

$spinner-color: $color-icon;
$spinner-background: $color-main;
$spinner-size: 30px;

.spinner-container {
  background-color: $spinner-background;
  border-radius: 0.5rem;
  border: none;
  outline: none;
  width: 54px;
  height: 54px;
}

.spinner {
  position: relative;
  top: calc(50% - #{$spinner-size / 2});
  left: calc(50% - #{$spinner-size / 2});
  width: $spinner-size;
  height: $spinner-size;
  animation: spinner 0.75s linear infinite;

  span {
    position: absolute;
    width: 100%;
    height: 100%;
    border-radius: 50%;
  }

  .spinner-inner-1 {
    background: linear-gradient(
      to right,
      rgba($spinner-color, 0) 0%,
      rgba($spinner-color, 0) 50%,
      $spinner-color 51%
    );
  }

  .spinner-inner-2 {
    background: linear-gradient(
      to top,
      rgba($spinner-background, 0) 0%,
      $spinner-background 100%
    );
  }

  .spinner-inner-3 {
    background: $spinner-background;
    $offset: $spinner-size * 0.1;
    top: $offset;
    left: $offset;
    width: $spinner-size - ($offset * 2);
    height: $spinner-size - ($offset * 2);
  }
}

@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
