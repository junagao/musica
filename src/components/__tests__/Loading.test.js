import React from 'react';
import { shallow } from 'enzyme';

import Loading from 'components/Loading';

describe('Loading', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Loading />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one div element with className of spinner-container and another with spinner', () => {
    expect(wrapper.find('div').length).toEqual(2);
    expect(
      wrapper
        .find('div')
        .at(0)
        .hasClass('spinner-container'),
    ).toEqual(true);
    expect(
      wrapper
        .find('div')
        .at(1)
        .hasClass('spinner'),
    ).toEqual(true);
  });

  it('should render three span elements with className of spinner-inner-1, spinner-inner-2, spinner-inner-3 respectively', () => {
    expect(wrapper.find('span').length).toEqual(3);
    expect(
      wrapper
        .find('span')
        .at(0)
        .hasClass('spinner-inner-1'),
    ).toEqual(true);
    expect(
      wrapper
        .find('span')
        .at(1)
        .hasClass('spinner-inner-2'),
    ).toEqual(true);
    expect(
      wrapper
        .find('span')
        .at(2)
        .hasClass('spinner-inner-3'),
    ).toEqual(true);
  });
});
