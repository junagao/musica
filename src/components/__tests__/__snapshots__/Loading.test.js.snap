// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<div
  className="spinner-container"
>
  <div
    className="spinner"
  >
    <span
      className="spinner-inner-1"
    />
    <span
      className="spinner-inner-2"
    />
    <span
      className="spinner-inner-3"
    />
  </div>
</div>
`;
